domain_utils/domain_utils.py

```diff
@@ -116,14 +116,61 @@ def hostname_subparts(url, include_ps=False, **kwargs):
     return subparts
 
 
-def get_stripped_url(url, scheme=False):
-    """Returns a url stripped to (scheme)?+hostname+path"""
+def get_stripped_url(url, scheme=False, non_http_scheme=None):
+    """
+    Returns a url stripped to (scheme)?+netloc+path
+    For example ``https://my.domain.net/a/path/to/a/file.html#anchor?a=1``
+    becomes ``my.domain.net/a/path/to/a/file.html``
+
+
+    URL parsing is done using std lib urllib.parse.urlparse
+    Using netloc means that a port is included, for example,
+    if it was in the path.
+    The method strips just the beginning and end being stripped.
+
+    :param url: URL to be parsed
+    :type url: str
+    :param scheme: If True, scheme will be prepended in
+        returned result, defaults to False
+    :type scheme: bool, optional
+    :param non_http_scheme: Action to take if scheme is not
+        ``http`` or ``https`` e.g. ``file:`` or ``about:blank``.
+        If None, return empty string.
+        If ``self``, return the original URL.
+        Default is None.
```

Comment:
`self` seems like a better default here. When running this on a database we'll probably have a bunch of non-http schemes and it seems better to have them map to URL rather than map down to None. It's also more intuitive: these non-http URLs can't be stripped down any further so we just return them as-is.

Reply:
Also since this only accepts two values I suggest using a boolean and renaming. Perhaps: `drop_non_http_urls` or something like that?

Reply:
I agree. Changing to self will change the default behavior (from crawl_utils). So if you're good with that, then great.

Like the boolean idea.

Reply:
I'm fine with that!